src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -99,6 +105,39 @@ def ensure_correct_owner(row: dict, role: str, obj_name: str) -> None:
         raise UnexpectedOwnerError(obj_name, role, actual_owner)
 
 
+def _get_paths_to_sync(
+    relative_files_to_sync: List[Path], deploy_root: Path, remote_paths: Set[str]
+) -> List[str]:
+    """Takes a list of paths that exist either locally or remotely, and returns a merged list of paths relative to the deploy root."""
+    paths = []
+    for file in relative_files_to_sync:
+        path = Path(deploy_root, file)
+        relpath = path.relative_to(deploy_root)
+        if not path.exists() and str(relpath) not in remote_paths:
+            raise FileError(
+                str(file), "This file does not exist either locally or remotely"
+            )
+        elif path.is_dir():
+            raise ClickException(f"Specifying directories is not supported: '{file}'")
+        else:
+            paths.append(str(relpath))
+    return paths
+
+
+def _map_paths_to_deploy_root(
+    paths: List[Path], artifacts: List[ArtifactMapping]
+) -> List[Path]:
+    """Maps a list of paths to being relative to deploy root."""
+    new_paths = []
+    for path in paths:
+        new_paths.append(determine_artifacts_file_path(path, artifacts))
+    return new_paths
+
+
+def _get_default_deploy_prune_value(files_argument: Optional[List[Path]]) -> bool:
```

Comment:
That's where I started with, but we have `version_processor` calling `sync_deploy_root_with_stage()` directly.

I preferred adding optional arguments and determine the default value here instead of resolving the default value in 2 different places. WDYT?

Reply:
I see okay, that makes sense.